byzcoin/transaction.go

```diff
@@ -231,6 +231,23 @@ func (instr Instruction) DeriveID(what string) InstanceID {
 	// strict domain separation now.
 }
 
+// DeriveIDArg calculates the id of the new instance for a spawn-instruction.
+// If the 'arg' is not present in the spawn-arguments,
+// it behaves like DeriveID(what).
+// If the instruction is not a spawn-instruction, it returns an error.
+func (instr Instruction) DeriveIDArg(what, arg string) (InstanceID, error) {
```

Comment:
Can't we use the `Argument` type direclty (without `Value`)?

```suggestion
func (instr Instruction) DeriveIDArg(what string, arg Argument) (InstanceID, error) {
```

Reply:
Problem is that the caller does not know where in the `instr.Spawn.Args` the correct argument is. He only knows the name of the argument...

Reply:
Yep, and the caller can create a "search "Argument`" as `Argument{Name: "my-name"}`.